src/renderer/backend/dx12.rs

```diff
@@ -750,6 +762,48 @@ impl TextureHeap {
             )
         })?;
 
+        self.device.CreateShaderResourceView(
+            &texture,
+            Some(&D3D12_SHADER_RESOURCE_VIEW_DESC {
+                Format: DXGI_FORMAT_R8G8B8A8_UNORM,
+                ViewDimension: D3D12_SRV_DIMENSION_TEXTURE2D,
+                Shader4ComponentMapping: D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING,
+                Anonymous: D3D12_SHADER_RESOURCE_VIEW_DESC_0 {
+                    Texture2D: D3D12_TEX2D_SRV {
+                        MostDetailedMip: 0,
+                        MipLevels: 1,
+                        PlaneSlice: Default::default(),
+                        ResourceMinLODClamp: Default::default(),
+                    },
+                },
+            }),
+            cpu_desc,
+        );
+
+        let id = TextureId::from(self.textures.len());
+        self.textures.push(Texture { resource: texture.clone(), gpu_desc, width, height });
+
+        Ok(id)
+    }
+
+    unsafe fn upload_texture(
+        &mut self,
+        texture_id: TextureId,
+        data: &[u8],
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let texture = &self.textures[texture_id.id()];
+        if texture.width != width || texture.height != height {
+            return Err(Error::new(
```

Comment:
Same observation as above, just marking the spot with a  comment for convenience.